packages/web-app/src/containers/transactionModals/DepositModal.tsx

```diff
@@ -48,20 +69,40 @@ const DepositModal: React.FC = () => {
   };
 
   const handleCtaClicked = useCallback(async () => {
-    const tokenInfo = await getTokenInfo(
-      input.tokenAddress,
-      provider,
-      CHAIN_METADATA[network].nativeCurrency
-    )
-    const amount = Number(input.amount) * Math.pow(10, tokenInfo.decimals)
-    const allowance = await tokenAllowance(input.tokenAddress)
-    if(allowance < amount) {
-      //TODO - Change modal label to approve
-      approve(input.tokenAddress, amount)
+    try {
+      setDepositProcessState(TransactionState.LOADING);
+      const tokenInfo = await getTokenInfo(
+        input.tokenAddress,
+        provider,
+        CHAIN_METADATA[network].nativeCurrency
+      )
+      const amount = Number(input.amount) * Math.pow(10, tokenInfo.decimals)
+      const allowance = await tokenAllowance(input.tokenAddress)
+      if (true || allowance < amount) {
+        setDepositProcessState(TransactionState.WAITING)
+        approve(input.tokenAddress, amount);
```

Comment:
We should check the tx returned right? approve is returning a tx await tx.wait() ??